### PR TITLE
Simplified storage index: removed config hash id in favor of using a unique model identifier

### PR DESCRIPTION
**Pull request by anton164**
This should simplify the use of the storage. All scripts and data was updated according.

Also updated the `store_summary_metrics` to behave as PUT request to ensure that other metrics are not overwritten.

## BEFORE CHANGE Example snippet for reading summaries and updating metrics
```
from sumtool.storage import get_summaries

generated_summary_lookup = {
  "BERTS2S": "2221cf6c3bad26c64a12417282dabf3c",
  "TranS2S": "adaab4ed6990eeae0ea8e98c78a722fe",
  "PtGen": "5641d358ff7a45cc65592938f1020675", 
  "TConvS2S": "dfadf4c1e1847488969892cc58be06c1"
}

# dictionary of doc id -> summary data
gold_summaries = get_summaries("xsum", "Gold")["dfd16d395e88b38829ff2d5b98743404"]

for model_name, model_id in data_lookup.items():
  # dictionary of doc id -> summary data
  model_generated_summaries = get_summaries("xsum", model_name)[model_id]

  # compute metrics, ...i.e. compare generated summaries with gold summaries 
  summary_metrics = compute_metrics(...)

  store_summary_metrics("xsum", model_name, model_id, summary_metrics)
```

## AFTER CHANGE Example snippet for reading summaries and updating metrics
```
from sumtool.storage import get_summaries

# dictionary of doc id -> summary data
gold_summaries = get_summaries("xsum", "maynez-gold")

for model_id in ["maynez-berts2s", "maynez-ptgen", ...]:
  # dictionary of doc id -> summary data
  model_generated_summaries = get_summaries("xsum", model_id)

  # compute metrics, ...i.e. compare generated summaries with gold summaries 
  summary_metrics = compute_metrics(...)

  store_summary_metrics("xsum", model_name, model_id, summary_metrics)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,18 +72,12 @@ flake8 sumtool/ interface/ scripts/
    - by loading them from an external dataset/paper ([example](scripts/store_xsum_annotated.py))
 2. Compute summary metrics for stored summaries using sumtool.
 
-![Storage Diagram](storage-diagram.jpg)
-
-#### `/data/<dataset>/<model-name>-summaries.json`
+#### `/data/<dataset>/<model-i>-summaries.json`
```
Comment (dleve123) [inline README.md:75]:
```suggestion
#### `/data/<dataset>/<model-id>-summaries.json`
```

File: data/xsum/data.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (dleve123) [inline data/xsum/data.txt:1]: I think this was just scratch work, should this be deleted?

File: interface/model_interface.py [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
-from sumtool import generate_xsum_summary, storage
+from sumtool import storage
 from backend.viz_data_loader import load_annotated_data_by_id
 import pandas as pd
 
 cache_summaries = storage.get_summaries("xsum", "facebook-bart-large-xsum")
-cache_summary_values = [*cache_summaries.values()][0]
-cache_keys = [x for x in [*cache_summary_values.keys()]]
+cache_keys = cache_summaries.keys()
+st.write()
```
Comment (dleve123) [inline interface/model_interface.py:8]: why the empty `st.write()`?

File: interface/model_interface.py [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
 annotated_data_by_id = load_annotated_data_by_id()
 filtered_annotated_data_by_id = {
-    k: annotated_data_by_id[k] for k in cache_keys if k in annotated_data_by_id
+    k: annotated_data_by_id[k]
+    for k in cache_summaries.keys()
+    if k in annotated_data_by_id
 }
 
 
-@st.experimental_singleton
```
Comment (dleve123) [inline interface/model_interface.py:15]: good catch 👍

File: sumtool/storage.py [hunk 5]
```diff
@@ -1,30 +1,61 @@
...
+    Strip leading and trailing whitespace, dashes, and underscores.
+
+    Taken from https://github.com/django/django/blob/master/django/utils/text.py
+
+    Args:
+        value: the string to slugify
+        allow_unicode: Convert to ASCII if 'allow_unicode' is False.
+
+    Returns:
+        the slugified strin
```
Comment (dleve123) [inline sumtool/storage.py:32]:
```suggestion
        the slugified string
```

File: sumtool/storage.py [hunk 6]
```diff
@@ -1,30 +1,61 @@
...
-def dataset_dir(dataset):
+def dataset_dir(dataset: str):
     return f"{STORAGE_DIR}/{dataset}"
 
 
+def model_path(dataset: str, model: str, ext):
+    return f"{dataset_dir(dataset)}/{slugify(model)}-{ext}"
+
+
+def slugify(value, allow_unicode=False):
```
Comment (dleve123) [inline sumtool/storage.py:18]: minor: seems like your adding types to the other functions, want to add them here?

File: sumtool/storage.py [hunk 7]
```diff
@@ -74,9 +105,7 @@ def store_model_summaries(
             stored_summaries = json.load(f)
 
     for document_id, summary in generated_summaries.items():
-        if model_config_hash not in stored_summaries:
-            stored_summaries[model_config_hash] = {}
```
Comment (dleve123) [inline sumtool/storage.py:78]: why this change?

File: data/xsum/facebook-bart-large-xsum-config.json [hunk 8]
```diff
@@ -82,7 +82,7 @@
   "static_position_embeddings": false,
   "student_decoder_layers": null,
   "student_encoder_layers": null,
-  "sumtool_name": "facebook/bart-large-xsum",
+  "sumtool_id": "facebook-bart-large-xsum",
```
Comment (dleve123) [inline data/xsum/facebook-bart-large-xsum-config.json:85]: so this `sumtool_id` is effectively the "foreign key" for the name of the metric, summary JSON files, correct? It _should_ correspond to the name of this config file, but it really could be anything? Is that correct?

**Review by dleve123 — CHANGES_REQUESTED** [review]
overall looks good! caught a couple of typos that we should fix before merging and left a question about some of the DB structures.

File: data/xsum/data.txt (hunk 2 above)
Comment (anton164) [reply data/xsum/data.txt:1]: Yes, good catch

File: sumtool/storage.py (hunk 7 above)
Comment (anton164) [reply sumtool/storage.py:78]: Because we no longer index summaries by the `model_config_hash`. They are indexed directly by the summary id.

**Comment by anton164** [comment]
Thanks for the review @dleve123 ! Your comments have been addressed

**Review by celine-lee — APPROVED** [review]

**Review by dleve123 — APPROVED** [review]

**Comment by anton164** [comment]
Blocked by #78 - will merge after that one to resolve conflicts (lmk when you merge it @dleve123)

**Comment by dleve123** [comment]
@anton164 #78 should be good to go once someone approves the changes I made in response to PR feedback.

**Comment by dleve123** [comment]
@anton164 I don't think we're seeing any git merge conflicts here, but I believe there's a semantic conflict to be addressed: removing/updating the main key in the metrics JSON documents ([example](https://github.com/factula/sumtool/blob/main/data/xsum/BERTS2S-metrics.json#L2)). Is that correct?

**Comment by anton164** [comment]
@dleve123 correct, I needed to migrate the entailment script & data to the updated API. Done now

**Pull request merged**